src/mozillavpn.cpp

```diff
@@ -1290,6 +1292,7 @@ void MozillaVPN::hardReset() {
 void MozillaVPN::hardResetAndQuit() {
   logger.debug() << "Hard reset and quit";
   hardReset();
+  connectionManager()->deleteTunnel();
```

Comment:
Should the inspector command be updated to do delete the tunnel?

```c++
  InspectorHandler::registerCommand(
      "quit", "Quit the app", 0,
      [](InspectorHandler*, const QList<QByteArray>&) {
        MozillaVPN::instance()->connectionManager()->quit();
        return QJsonObject();
      });
```

Reply:
This is when the app closes, right? If so, I don't think we want this here. 

On iOS and Android, removing the tunnel config means the user is going to get a system prompt the next time they turn on the VPN. I believe we want to only remove this when a user is signed out. But perhaps I'm misunderstanding what the quit action does here?

Reply:
```c++
void ConnectionManager::quit() {
  logger.debug() << "Quitting";

  if (m_state == StateInitializing || m_state == StateOff) {
    m_nextStep = Quit;
    emit readyToQuit();
    return;
  }

  m_nextStep = Quit;

  if (m_state == StateOn || m_state == StateSwitching ||
      m_state == StateSilentSwitching || m_state == StateConnecting ||
      m_state == StateCheckSubscription) {
    deactivate();
    return;
  }
}
```

We are deactivating the VPN on quit(), I guess I assumed we would want to remove any lingering process upon deactivation but if that assumption is incorrect we can discard this comment

Reply:
Also, I'm open to better names then `deleteTunnel`. Maybe `deleteTunnelSystemConfig`? That's too wordy maybe... I first went with `removeTunnel`, but I was worried that sounded too much like breaking down the tunnel - and I wanted to make it clearly destructive/rare. `deleteOSTunnelConfig`?

Reply:
I'd hope we're killing any lingering processes elsewhere in the code. 😬  This PR is about what is saved to disk for the VPN config, so I wasn't focused on the active VPN breakdown (though I did test to make sure the VPN status didn't affect this deletion).

Reply:
I agree `deleteTunnel` and `removeTunnel` are both a bit misleading, it definitely confused me 😁 I don't think `deleteTunnelSystemConfig` is too wordy. Both that and `deleteOSTunnelConfig` are reasonable names IMO

Reply:
I'm going with `deleteOSTunnelConfig` for now - about to update this. Let me know if you prefer something else.